src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs

```diff
@@ -77,6 +77,7 @@ public AssemblyName()
             }
         }
 
+        [Obsolete(Obsoletions.ProcessorArchitectureMessage, DiagnosticId = Obsoletions.ProcessorArchitectureDiagId, UrlFormat = Obsoletions.SharedUrlFormat)]
```

Comment:
@i3arnon the attribute should be added to the ref too https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/ref/System.Runtime.cs